tests/nmap_agent_test.py

```diff
@@ -225,10 +262,85 @@
 """
 
 
+def testNmapAgentLifecycle_whenIpv6WithHostBits_shouldReport(
+    nmap_test_agent: nmap_agent.NmapAgent,
+    agent_mock: List[message.Message],
+    ipv6_msg: message.Message,
+    mocker: plugin.MockerFixture,
+) -> None:
+    """Unit test of nmap agent when ipv6 with host bits is provided."""
+    mocker.patch(
+        "agent.nmap_wrapper.NmapWrapper.scan_hosts",
+        return_value=(IPV6_JSON_OUTPUT, IPV6_HUMAN_OUTPUT),
+    )
+
+    nmap_test_agent.process(ipv6_msg)
+
+    assert len(agent_mock) == 2
+    assert agent_mock[0].selector == "v3.asset.ip.v6.port.service"
+    assert agent_mock[1].selector == "v3.report.vulnerability"
+    assert agent_mock[1].data["risk_rating"] == "INFO"
+    assert agent_mock[1].data["title"] == "Network Port Scan"
+    assert (
+        "2600:3c01:224a:6e00:f03c:91ff:fe18:bb2f"
+        in agent_mock[1].data["technical_detail"]
+    )
+
+
+def testNmapAgent_whenInvalidIpv6_shouldRaiseAnError(
+    nmap_test_agent: nmap_agent.NmapAgent,
+    agent_mock: List[message.Message],
+    invalid_ipv6_msg: message.Message,
+) -> None:
+    """Test that invalid IPv6 raises an AddressValueError."""
+    with pytest.raises(ipaddress.AddressValueError, match="At least 3 parts expected"):
+        nmap_test_agent.process(invalid_ipv6_msg)
+
+    # Ensure no messages were emitted
+    assert len(agent_mock) == 0
```

Comment:
Add additional assertions to confirm that no other unexpected side effects occur when an error happens.